### DISCUSSION TITLE
Text best fitting into a fixed size box

### DISCUSSION THREAD

**Discussion by tbmuc** [ImageMagick/ImageMagick#4311] [Help]
Hello, 
how to make a text best fitting into a fixed size box. It should use either 100% of the length or width of the box but not be stretched. The current souloution does not work correctly:

This is the part of the code which generates the text:

( -size 4544x832 -font "C:\\Users\\Thomas\\Google Drive\\POD\\imagemagic\\mkcanvas\\fonts\\Billy Scriptian.ttf" -background None -fill #3b3734 caption:%text% -trim -resize 2272x416 -gravity center -extent 2272x416 ) 

![henry](https://user-images.githubusercontent.com/48103248/135753308-ecfb1112-aa24-4419-b697-10baf956c3a3.jpg)

**Comment by snibgo** [comment]
What version of IM, on what platform?

> how to make a text best fitting into a fixed size box. It should use either 100% of the length or width of the box but not be stretched.

If the "fixed size box" has width W and height H, then...
```
magick -size WxH caption:Hello out.png
```
... will choose a font size so the text has width W or height H. However, it will leave a small white space around the width or height, and a larger white space in the other dimension. If you don't want the small white space, you could fill this by resizing, but you say "but not be stretched."

So I am not sure what you want.

**Comment by tbmuc** [comment]
I have it tried with the command: magick -size WxH caption:Hello out.png

The problem is that the y is cut:

![out](https://user-images.githubusercontent.com/48103248/135762266-ebe2d57a-39ae-4468-b51c-bf45819d7d62.png)

This is the font:
https://www.dafont.com/billy-scriptian.font

**Comment by snibgo** [comment]
I suspect a problem in the font file. Note the following:
```
magick -debug annotate -size 300x100 -background #0ff -pointsize 50 -undercolor #ff0 -font "Billy Scriptian.ttf" caption:Henry x.png
```
The "y" extends below the yellow undercolor.
![x](https://user-images.githubusercontent.com/33812368/135763615-3f38b5e9-4fc7-438e-b23b-0abdec67b000.png)

**Comment by fmw42** [comment]
The font glyphs are not accounting for the drop in the y. Try to find another font from a different company or change to a totally different font. I do not think this is an ImageMagick issue.

**Comment by tbmuc** [comment]
@snibgo @fmw42 Thank you very much for the hints. Do you know a font app which can correct the issue?

**Reply by fmw42** [reply]
Sorry, no. I do not know of any font editors. You can try by searching Google. Perhaps you will find one for your OS.

**Reply by fmw42** [reply]
I suggest you just get a different font.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT main branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by tbmuc** [comment]
Hello Cristy,

thank you very much. Are there already windows x64 binaries with the patch available. If so, where are they. If not, how to build them.

Thanks 
Thomas

**Comment by urban-warrior** [comment]
See building ImageMagick from source from the ImageMagick web site.  Otherwise there will be an official release, including Windows binary distributions, within a week or two.
